playwright/src/test/java/com/microsoft/playwright/TestElementHandleSelectText.java

```diff
@@ -0,0 +1,91 @@
+package com.microsoft.playwright;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TestElementHandleSelectText extends TestBase {
+
+  @Test
+  void shouldSelectTextarea() {
+    page.navigate(server.PREFIX + "/input/textarea.html");
+    ElementHandle textarea = page.querySelector("textarea");
+    textarea.evaluate("textarea => textarea.value = 'some value'");
+    textarea.selectText();
+    if (isFirefox()) {
+      assertEquals(0, textarea.evaluate("el => el.selectionStart"));
+      assertEquals(10, textarea.evaluate("el => el.selectionEnd"));
+    } else {
+      assertEquals("some value", page.evaluate("() => window.getSelection().toString()"));
+    }
+  }
+
+  @Test
+  void shouldSelectInput() {
+    page.navigate(server.PREFIX + "/input/textarea.html");
+    ElementHandle input = page.querySelector("input");
+    input.evaluate("input => input.value = 'some value'");
+    input.selectText();
+    if (isFirefox()) {
+      assertEquals(0, input.evaluate("el => el.selectionStart"));
+      assertEquals(10, input.evaluate("el => el.selectionEnd"));
+    } else {
+      assertEquals("some value", page.evaluate("() => window.getSelection().toString()"));
+    }
+  }
+
+  @Test
+  void shouldSelectPlainDiv() {
+    page.navigate(server.PREFIX + "/input/textarea.html");
+    ElementHandle div = page.querySelector("div.plain");
+    div.selectText();
+    assertEquals("Plain div", page.evaluate("() => window.getSelection().toString()"));
+  }
+
+  @Test
+  void shouldTimeoutWaitingForInvisibleElement() {
+    page.navigate(server.PREFIX + "/input/textarea.html");
+    ElementHandle textarea = page.querySelector("textarea");
+    textarea.evaluate("e => e.style.display = 'none'");
+    try {
+      textarea.selectText(new ElementHandle.SelectTextOptions().withTimeout(3000));
+      fail("did not throw");
+    } catch (PlaywrightException e) {
+      assertTrue(e.getMessage().contains("element is not visible"));
+    }
+  }
+
+  @Test
+  void shouldWaitForVisible() {
+    page.navigate(server.PREFIX + "/input/textarea.html");
+    ElementHandle textarea = page.querySelector("textarea");
+    textarea.evaluate("textarea => textarea.value = 'some value'");
+    textarea.evaluate("e => e.style.display = 'none'");
+
+    AtomicBoolean done = new AtomicBoolean(false);
+    // Async look for element timeout 3 seconds and return status if found or not
+    CompletableFuture<Boolean> completableFuture = CompletableFuture.supplyAsync(() -> {
```

Comment:
Playwright is not thread safe, it's designed to work on a single thread. I'll probably add a check that the APIs are called only from one thread.

These kinds of tests (should wait for ..) are essentially testing backend behavior. In typescript they are very easy to implement with promises but it's much more difficult with sync API. Given that there is a good coverage for the functionality upstream I was skipping such tests in java. Alternatively, if you really want to port such tests they can be converted to trigger the condition asynchronously from inside the page but in this case I suggest you drop this test in java (just leave an empty method similar to `shouldWaitForDisplayNoneToBeGone`)

Reply:
OK got it.  I will skip the last test case